Source/Urho3D/Utility/GLTFImporter.cpp

```diff
@@ -3568,10 +3568,13 @@ class GLTFSceneImporter : public NonCopyable
         staticModel.SetModel(model);
         staticModel.SetCastShadows(true);
 
-        const StringVector& materialList = modelImporter_.GetModelMaterials(meshIndex, skinIndex);
-        for (unsigned i = 0; i < materialList.size(); ++i)
+        const ResourceRefList& materialList = modelImporter_.GetModelMaterials(meshIndex, skinIndex);
+        if (materialList.type_ != Material::GetTypeStatic())
+            return;
+
+        for (unsigned i = 0; i < materialList.names_.size(); ++i)
         {
-            auto material = cache->GetResource<Material>(materialList[i]);
+            auto material = cache->GetResource<Material>(materialList.names_[i]);
             staticModel.SetMaterial(i, material);
```

Comment:
You can just call SetMaterialAttr instead of running through the loop here. The whole code block could be replaced with one call to SetMaterialAttr

Reply:
Actually please check that it would work. May need to call ApplyAttributes or keep it as is. @rokups ?

Reply:
I was skeptical about changing it to SetMaterialAttr because I don't know if it would work at runtime. Are tests implemented for that?